### PR TITLE
Refactor Confd Yang Modules Setup Script

### PR DESCRIPTION
**Pull request by ahmkar94**
These changes aim to improve the modularity and maintainability of our codebase, making it easier to manage and extend the YANG module setup process in the future.

### CODE REVIEW

File: utils/yang_modules_setup.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    fi
+}
+
+MODULES_LIST_FILE_CONFD=${MODULES_LIST_FILE:-$(dirname "$0")/yang_modules_confd.inc}
+MODULES_LIST_FILE_CONTAINTERS=${MODULES_LIST_FILE:-$(dirname "$0")/yang_modules_containers.inc}
+MODULES_LIST_FILE_INTERFACES=${MODULES_LIST_FILE:-$(dirname "$0")/yang_modules_interfaces.inc}
+
+check_file_exists $MODULES_LIST_FILE_CONFD
+check_file_exists $MODULES_LIST_FILE_CONTAINTERS
+check_file_exists $MODULES_LIST_FILE_INTERFACES
```
Comment (mattiaswal) [inline utils/yang_modules_setup.sh:26]: The file should be as an input to the script example:
```
./yang_modules_setup.sh file-with-modules
```
And in confd, just call the script twice
```
yang_modules_setup.sh confd.inc
yang_modules_setup.sh interfaces.inc
```

File: utils/yang_modules_setup.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+check_file_exists() {
+    file=$1
+
+    if [ ! -f "$file" ]; then
+        echo "Modules list file not found: $file"
+        exit 1
+    fi
+}
+
+MODULES_LIST_FILE_CONFD=${MODULES_LIST_FILE:-$(dirname "$0")/yang_modules_confd.inc}
```
Comment (mattiaswal) [inline utils/yang_modules_setup.sh:20]: Didn't it work to have a variable called MODULES as before and use source on the file requested?

**Review by mattiaswal — CHANGES_REQUESTED** [review]

**Review by mattiaswal — APPROVED** [review]
Great work :fireworks:  :+1:

**Comment by troglobit** [comment]
I'm not really happy with all the whitespace changes in this PR.  Lots of initial indent introduced in the .mk file that does not comply with Buildroot standards.

Also, I'm left a confounded with the move of logic from src/confd to package/confd.  Why can't we keep the internals of which modules are needed in src/confd?

**Comment by mattiaswal** [comment]
> I'm not really happy with all the whitespace changes in this PR. Lots of initial indent introduced in the .mk file that does not comply with Buildroot standards.
> 
> Also, I'm left a confounded with the move of logic from src/confd to package/confd. Why can't we keep the internals of which modules are needed in src/confd?

The logic, what do you mean? The script itself has been moved to utils/ to make it nicer when calling from an external build, also remove all knowledge about containers etc from that feels right, with that in mind. The calling of the script has always been in confd.mk. Is it the .inc files you do not like in src/confd/, I can agree, it should be moved, it should be with the source code.

**Comment by ahmkar94** [comment]
I had a feeling that I shouldn't touch the indentations. I just wasn't used to see 'if' logic without indentations xD
But I'll make a quick followup commit (PR) on that, which will also include a minor update in the list for the yang_modules_confd.inc file (missing a module).

**Comment by troglobit** [comment]
We've had parts of confd spread out all over the tree before, and it was very difficult to keep track of origin and use.  So we gathered the pieces and put them back into src/confd/ where they belong.

Think of it like this: if confd was an external project (not living in src/) with a base set of YANG models and tools to manage it, we'd use package/confd/ to integrate it into Infix and any Infix extensions would live in board/common/.

So yes, the .inc files should live in src/confd/, next setup.sh in the scripts/ sub-directory.

**Comment by ahmkar94** [comment]
Created a followup PR https://github.com/kernelkit/infix/pull/487

**Comment by troglobit** [comment]
FYI: just merged main into my branch ... and I had zero (0!) conflicts?  Took me a while to realize, but in this PR, you forgot to remove the old `confd/scripts/setup.sh` :smile: 

No worries, I'll migrate my local changes of that script to the new and remove `confd/scripts/setup.sh` on my branch.

**Comment by ahmkar94** [comment]
> FYI: just merged main into my branch ... and I had zero (0!) conflicts? Took me a while to realize, but in this PR, you forgot to remove the old `confd/scripts/setup.sh` 😄
> 
> No worries, I'll migrate my local changes of that script to the new and remove `confd/scripts/setup.sh` on my branch.

Ohhhh... 
Thnx :-)

**Pull request merged**